libevmasm/Assembly.cpp

```diff
@@ -756,3 +759,31 @@ LinkerObject const& Assembly::assemble() const
 	}
 	return ret;
 }
+
+vector<size_t> Assembly::decodeSubIds(u256 _data)
+{
+	size_t length = static_cast<size_t>((_data >> 248) & 0xFF);
+	if (length == 0)
+		length = 1;
+	std::vector<size_t> subIds;
+	u256 const mask = 0xFF;
+	while (length--)
+	{
+		auto currentSubId = static_cast<size_t>(_data & mask);
+		subIds.emplace_back(currentSubId);
+		_data >>= 8;
+	}
+	return subIds;
```

Comment:
I think this can be simplified a bit:
```suggestion
	for (size_t reverseSubIndex = 0; reverseSubIndex < length - 1; ++reverseSubIndex)
		subIds.emplace_back(static_cast<size_t>((_data >> reverseSubIndex * 8) & 0xff));
```